src/strace.c

```diff
@@ -1554,8 +1558,12 @@ exec_or_die(void)
 		 * It is important to set groups before we
 		 * lose privileges on setuid.
 		 */
-		if (initgroups(username, run_gid) < 0)
-			perror_msg_and_die("initgroups");
+		// TODO: provide a way to pass "setgroups()" data to avoid
+		// the initgroups() that requires libnss
+		if (cmdline_uid == -1 && cmdline_gid == -1)  {
+			if (initgroups(username, run_gid) < 0)
+				perror_msg_and_die("initgroups");
+		}
```

Comment:
If it cannot call initgroups it has to invoke setgroups with an empty list just to avoid leaking groups.